scripts/rm_synthesis.py

```diff
@@ -0,0 +1,371 @@
+    os.remove(ascii_archive) #remove the archive file, we don't need it anymore
+
+    #find the phase range(s) to fit:
+    if not phase_ranges:
+        phase_ranges = find_on_pulse_ranges(I, **kwargs_gfit)
+    logger.info("Using phase ranges: {}".format(phase_ranges))
+
+    #run rmfit with -w option
+    rm_dict = {}
+    for i, phase_min, phase_max in zip(range(len(phase_ranges)//2), phase_ranges[0::2], phase_ranges[1::2]):
+        logger.info("Performing RM synthesis on phase range: {0} - {1}".format(phase_min, phase_max))
+        rm_dict[str(i)] = {}
+        rmfit_quad(archive, phase_min, phase_max)
+        #read the QUVflux.out file
+        QUVflux = "QUVflux.out"
+        rmfreq, rmI, rmI_e, rmQ, rmQ_e, rmU, rmU_e, = read_rmfit_QUVflux(QUVflux)
+    
+        #make plot name if needed
+        if plot:
+            plotname = "{}_".format(label)
+            if len(phase_ranges)>2:
+                plotname += "{0}_".format(i)
+            plotname += "RMsynthesis.png"
+            kwargs_rms["plotname"] = plotname
+        else:
+            plotname = None
+        kwargs_rms["phase_range"] = (phase_min, phase_max)
+
+        #perform RM synthesis
+        rm, rm_e = IQU_rm_synth(rmfreq, rmI, rmI_e, rmQ, rmQ_e, rmU, rmU_e, **kwargs_rms)
+        rm_dict[str(i)]["rm"]           = rm
+        rm_dict[str(i)]["rm_e"]         = rm_e
+        rm_dict[str(i)]["phase_range"]  = (phase_min, phase_max)
+        rm_dict[str(i)]["plotname"]     = plotname
+        rm_dict[str(i)]["label"]        = label
+
+        #move plot out of workign dir
+        if plotname:
+            os.rename(plotname, "../{}".format(plotname))
+
+        #keep quvflux if needed
+        if keep_QUV:
+            quvflux_name = label
+            if len(phase_ranges)>2:
+                quvflux_name += "_{}".format(i)
+            quvflux_name += "_QUVflux.out"
+            os.rename("QUVflux.out", "../{}".format(quvflux_name))
+
+    if write:
+        filename = "{}_".format(label)
+        filename += "RMsynthesis.txt"
+        write_rm_to_file(filename, rm_dict)
+        os.rename(filename, "../{}".format(filename))
+    else:
+        filename = None
+    
+    os.chdir("../")
+    allfiles = glob.glob("{}/*".format(randomgen))
+    for afile in allfiles:
+        os.remove(afile)
+    os.rmdir("{}".format(randomgen))
+
+    return rm_dict, filename
+
+if __name__ == '__main__':
+  
+    loglevels = dict(DEBUG=logging.DEBUG,
+                     INFO=logging.INFO,
+                     WARNING=logging.WARNING,
+                     ERROR = logging.ERROR)
+
+    parser = argparse.ArgumentParser(description="A script that performs the RM synthesis technique",\
+                                    formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    required = parser.add_argument_group("Required Inputs:")
+    required.add_argument("-f", "--archive", type=str, help="The name of the archive (.ar) file to work with")
```

Comment:
You can use the add "required=True" to force this option to be used

Reply:
Fixed